sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -709,7 +694,7 @@ static TableFieldSchema tableFieldSchemaFromDescriptorField(
     if (fieldDescriptor.isRequired()) {
       tableFieldSchemaBuilder = tableFieldSchemaBuilder.setMode(TableFieldSchema.Mode.REQUIRED);
     }
-    if (fieldDescriptor.isOptional()) {
+    if (fieldDescriptor.isOptional() || !fieldDescriptor.isRequired()) {
       tableFieldSchemaBuilder = tableFieldSchemaBuilder.setMode(TableFieldSchema.Mode.NULLABLE);
     }
```

Comment:
Wouldn't this overwrite a REPEATED mode to NULLABLE? Maybe we don't need this extra check, according to [docs](https://cloud.google.com/java/docs/reference/google-cloud-bigquerystorage/latest/com.google.cloud.bigquery.storage.v1beta2.TableFieldSchema.Builder#com_google_cloud_bigquery_storage_v1beta2_TableFieldSchema_Builder_setMode_com_google_cloud_bigquery_storage_v1beta2_TableFieldSchema_Mode_), the mode already defaults to nullable.

Reply:
Alternatively, we could keep this check but turn these into else-if conditions

Reply:
made these else/if